tornado/web.py

```diff
@@ -179,20 +179,34 @@ def __init__(self, application, request, **kwargs):
         self._headers = None  # type: httputil.HTTPHeaders
         self.path_args = None
         self.path_kwargs = None
-        self.ui = ObjectDict((n, self._ui_method(m)) for n, m in
-                             application.ui_methods.items())
-        # UIModules are available as both `modules` and `_tt_modules` in the
-        # template namespace.  Historically only `modules` was available
-        # but could be clobbered by user additions to the namespace.
-        # The template {% module %} directive looks in `_tt_modules` to avoid
-        # possible conflicts.
-        self.ui["_tt_modules"] = _UIModuleNamespace(self,
-                                                    application.ui_modules)
-        self.ui["modules"] = self.ui["_tt_modules"]
```

Comment:
Don't you have to initialize self._ui here?

Reply:
clear does it

Reply:
Sorry, I missed that.